Sources/IRGen/Preprocessor/IRPreprocessor.swift

```diff
@@ -49,28 +58,16 @@ public struct IRPreprocessor: ASTPass {
   }
 
   // MARK: Declaration
-  public func process(structDeclaration: StructDeclaration,
-                      passContext: ASTPassContext) -> ASTPassResult<StructDeclaration> {
-    let environment = passContext.environment!
-    var structDeclaration = structDeclaration
-
-    let conformingFunctions = environment.conformingFunctions(in: structDeclaration.identifier.name)
-      .compactMap { functionInformation -> StructMember in
-        var functionDeclaration = functionInformation.declaration
-        functionDeclaration.scopeContext = ScopeContext()
-
-        return .functionDeclaration(functionDeclaration)
-      }
-
-    structDeclaration.members += conformingFunctions
-
-    return ASTPassResult(element: structDeclaration, diagnostics: [], passContext: passContext)
-  }
-
   public func process(variableDeclaration: VariableDeclaration,
                       passContext: ASTPassContext) -> ASTPassResult<VariableDeclaration> {
+    var variableDeclaration = variableDeclaration
     var passContext = passContext
 
+    if let structDeclarationContext = passContext.structDeclarationContext,
```

Comment:
How come we are still dealing with `Self` in this pass?

Reply:
good spot! not sure, I'll investigate